Automattic/jetpack
# AI Extension: start to extend Jetpack Form block

This PR extends the Jetpack Form block with AI Assistant. On this PR:

* Add a `isPossibleToExtendJetpackFormBlock()` to take over to know when it's possible to extend
* Add `AiAssistantToobarButton` component to show in the toolbar
* Add a withAiAssistantToolbarButton HOC to extend the BlockEdit function of the Jetpack form component

Part of https://github.com/Automattic/jetpack/issues/32160
Fixes #

## Proposed changes:

* AI Extension: start to extend Jetpack From block

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Test in different environments / infrastructures (Simple, Atomic and Jetpack)

* Go to your testing site
* Ensure the `beta` extension is not enabled
* Create Jetpack Form block instance
* Confirm the toolbar doesn't contain the AI Assistant button
* Enable `beta` extension
* Confirm the toolbar now shows a simple AI button

<img width="675" alt="Screenshot 2023-08-01 at 12 15 49" src="https://github.com/Automattic/jetpack/assets/77539/55f19253-9dab-4b04-8834-af9c7c91dae5">
